lib/client.js

```diff
@@ -866,15 +866,10 @@ MqttClient.prototype._handlePubrel = function (packet, callback) {
     that = this;
 
   that.incomingStore.get(packet, function (err, pub) {
-    if (err) {
-      return that.emit('error', err);
-    }
-
-    if ('pubrel' !== pub.cmd) {
+    if (!err && 'pubrel' !== pub.cmd) {
       that.emit('message', pub.topic, pub.payload, pub);
-      that.incomingStore.put(packet);
     }
-
+    that.incomingStore.put(packet);
```

Comment:
I'm not really convinced by this, I think this should be a `incomingStore.del()` in the first place.